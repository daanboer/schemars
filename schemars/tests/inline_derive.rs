mod util;
use schemars::JsonSchema;
use util::*;

#[allow(dead_code)]
#[derive(JsonSchema)]
struct Parent {
    child: Child,
}

#[allow(dead_code)]
#[derive(JsonSchema)]
#[schemars(inline)]
struct Child {
    name: String,
    age: u16,
}

#[test]
fn test_non_referenceable_schema() -> TestResult {
    test_default_generated_schema::<Parent>("inline_derive")
}
