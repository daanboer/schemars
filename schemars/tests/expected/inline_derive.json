{
  "$schema": "http://json-schema.org/draft-07/schema#",
  "title": "Parent",
  "type": "object",
  "required": [
    "child"
  ],
  "properties": {
    "child": {
      "type": "object",
      "required": [
        "name", 
        "age"
      ],
      "properties": {
        "name": {
          "type": "string"
        },
        "age": {
          "type": "integer",
          "minimum": 0,
          "format": "uint16"
        }
      }
    }
  }
}